### DISCUSSION TITLE
Can Pretix handle multi-day events with different arrivals and departures, pricings, accomodations, and capacities

### DISCUSSION THREAD

**Discussion by mielp** [pretix/pretix#4411] [Q&A]
Hi,

I am evaluating Pretix to replace the registration system for a nonprofit I'm volunteering for.

We are currently using online forms, maintaining a large spreadsheet with 15 tabs, and sending confirmation emails manually to each participant (+ adding them to this year's participants mailing list).

I did test a few Pretix features today and I think it can save us a lot of work, and especially many of the features are very useful, like invoice generation, multilingual forms, , . but we do have a pretty complex registration system today, by choice, and I have not yet found out if some or all of the needs in our current situation are supported:

- **Stay duration:**
  - We hold a single event a year, that lasts over 10+ days
  - Participants can book a ticket that indicated when they arrive and when they leave
  - They can mention their desired stay when registering for the waiting list
  - We may have check-in days, so the choice for arrival day is limited (but not the choice for departure day)
  - Participants often ask us to change their arrival and departure days, including on site during the event
- **Pricing:**
  - The ticket price includes a fixed part (membership fee) plus a daily rate depending on how long the participants are staying
  - The participant can ask for a discount ticket or for a support ticket, which change the membership fee and daily rate (-30% and +30% of the base fee and rate)
  - Participants who are invited as speakers get a 1 day discount (if they stay for 5 days, they pay 4)
  - Participants can decide to make a donation (and receive a donation receipt along with their invoice)
- **Capacity:**
  - We have an overall capacity of about 100 participants simultaneously on site
  - This means that if a participant leaves on the 6th day, there is capacity for another participant to arrive on the same day
  - Also, if a participant arrives on the 4th day, there is capacity for another participant to arrive earlier if they leave on the 4th or before
- **Accommodation:**
  - Participants can choose their accommodation type: bed, tent, or no accommodation
  - Participants may benefit from a 15% reduction based on the chosen accommodation type
  - Each accommodation type has different capacity (for instance 70 beds and 30 tents, no limit for no accommodation)
- **Waiting list:**
  - Some participants on the waiting list have priority depending on their answers to the registration form
  - Priority in our case means that some percentage of seats are reserved for these participants, even if they register at a later date than non-priority participants on the waiting list; therefore the two waiting lists are moving at different "speeds"
  - In other words, the capacity for priority participants is different from the capacity for non-priority participants
- **Check-in:**
  - More than half of the participants usually pay on site with cash, and often make small donations at that time ("keep the change")
  - A few registrations are done on site at check-in, even though the online registration closes weeks before the event
- **Registration:**
  - After registering successfully (no longer on the waiting list), the user is asked follow-up questions (specific needs, emergency contact, languages spoken, diet, and so on)

This looks like a huge shopping list, but this is our current situation. We have some room to change these requirements to adapt to the software, and we may be okay maintaining a hybrid system with some spreadsheets to support additional use cases, but some are also deal breakers for us.

I believe it's possible to make most of this using quotas, automatic discounts, vouchers, and maybe different products and product variations, but I haven't grasped how to do all that yet, and I'm not even sure whether to create a singular event (since that's what it is) or an event series (since it's over several days).

I guess if you have some idea on how to set up some of these functionalities, it can help us (and maybe others!) decide.

My thought for now is to create a multi-day event, make the membership fee and the accommodations be mandatory add-on product categories, and to have the discount/support pricing proposed in the form of discounts, but I'm not sure it can really represent our registration system, or even its pricing rules.

Do you have suggestions on how to do this or feedback on what's possible or not?

**Comment by raphaelm** [comment]
Giving a full answer would probably require 1h+ to fully think this through, which is more than I can spend here on the forum.

My gut feeling is: You will only be able to solve this in pretix with severe limitations, and the user experience will probably not be great.

The capacity handling should work when creating every day as an (add-on) product, however that would mean people not only choose their arrival and departure, but they'd manually need to select every day in between as well. Not a great UX, either for bookings nor for later changes. Discount and support tickets could be variations of the products, I'd rather not handle it as discounts (discounts do not allow a choice). The waiting list would need to be handled outside of pretix.
